pipeline/load.py

```diff
@@ -57,14 +58,104 @@ def load_artists(new_artist, db_connection: extensions.connection):
+        new_item['title'] = new_item['title'].replace("'", "''")
+        cur.execute(f"""INSERT INTO item(item_type_id, item_name, artist_id, item_image) 
+                    VALUES ({item_type_id}, '{new_item['title']}', {artist_id}, '{new_item['image']}')""")
+        db_connection.commit()
+        
+def load_item_genres(new_item, db_connection):
+    """
+    Connects to items to the genres.
+    """
+    with db_connection.cursor() as cur:
+        new_item['title'] = new_item['title'].replace("'", "''")
+        cur.execute(f"SELECT item_id FROM item WHERE item_name = '{new_item['title']}'")
+        item_id = cur.fetchone()[0]
+        
+        cur.execute(f"SELECT genre_id FROM genre WHERE genre='{new_item['tags'].lower()}'")
+        genre_id = cur.fetchone()[0]
+
+        cur.execute(f"SELECT * FROM item_genre;")
+        item_genres = cur.fetchall()
+        for item_genre in item_genres:
+            if item_genre[1] == item_id and item_genre[2] == genre_id:
+                return 
+    
+
+        cur.execute(f"INSERT INTO item_genre(item_id, genre_id) VALUES ({item_id},{genre_id})")
+        db_connection.commit()
+
+def load_sales_event(new_sale, db_connection):
+    with db_connection.cursor() as cur:
+        cur.execute(f"SELECT country_id FROM country WHERE country='{new_sale['country']}';")
+        country_id = cur.fetchone()[0]
+
+        new_sale['title'] = new_sale['title'].replace("'", "''")
+        cur.execute(f"SELECT item_id FROM item WHERE item_name='{new_sale['title']}'")
+        item_id = cur.fetchone()[0]
+
+        cur.execute(f"""INSERT INTO sale_event(sale_time, amount, country_id, item_id) 
+                    VALUES ('{new_sale['at']}', {new_sale['amount_paid_usd']}, {country_id}, {item_id}) """)
+        db_connection.commit()
+
+def load_data(sale_df_flat_tags: pd.DataFrame, sale_df: pd.DataFrame, con: extensions.connection):
+    sale_df_flat_tags['tags'].apply(load_genres, db_connection=con)
+    print("Tags added")
+    sale_df_flat_tags['artist'].apply(load_artists, db_connection=con)
+    print("Artists added")
+    sale_df_flat_tags['country'].apply(load_countries, db_connection=con)
+    print("Countries added")
+    sale_df_flat_tags.apply(load_items, db_connection=con, axis=1)
+    print("Items added")
+    sale_df_flat_tags.apply(load_item_genres, db_connection=con, axis=1)
+    print("Item genres added")
+    sale_df.apply(load_sales_event, db_connection=con, axis=1)
+    print("Sales added")
 
 if __name__ == "__main__":
     load_dotenv()
     music_df = pd.read_csv("clean_data.csv")
+    music_df_not_clean_tags = pd.read_csv("clean_data_no_tags.csv")
+
+    connection = get_db_connection()
+    start = perf_counter()
+
+    load_data(music_df, music_df_not_clean_tags, connection)
 
-    con = get_db_connection()
-    music_df['tags'].apply(load_genres, db_connection=con)
-    music_df['artist'].apply(load_artists, db_connection=con)
+    print(f"Time taken: {perf_counter() - start}")
```

Comment:
Is this part still needed? 🧐 It reads csv files that aren't in the repo anymore too I think